ij-server/src/main/kotlin/org/jetbrains/research/ij/headless/server/PsiFileManager.kt

```diff
@@ -24,44 +27,51 @@ class PsiFileManager(private val templatesDirPath: Path?) {
         private val disposable: Disposable
     )
 
-    private val idToLanguage = mutableMapOf<String, Language>()
-    private val singleFileProjects = mutableMapOf<String, SingleFileProject>()
+    private val singleFileProject = run {
+        logger.info("Start to create a new project for language \"${language.id}\"")
 
-    fun getLanguageById(languageId: String) = idToLanguage.getOrPut(languageId) {
-        Language.findLanguageByID(languageId) ?: error("No such language by id $languageId")
-    }
+        val (project, disposable) = createProject(
+            language,
+            templatesDirPath?.resolve(language.id.lowercase()) ?: Path.of(".")
+        ) ?: run {
+            logger.error("Can not create a project for language \"${language.id}\"")
+            error("Can not create a project for language \"${language.id}\"")
+        }
 
-    fun initSingleFileProject(languageId: String) {
-        singleFileProjects.getOrPut(languageId) {
-            val language = getLanguageById(languageId)
-            logger.info("Start to create new psi project for language $language")
-
-            val (project, disposable) = createProject(
-                language,
-                templatesDirPath?.resolve(language.id.lowercase()) ?: Path.of(".")
-            ) ?: run {
-                logger.error("Can not create project for language $language")
-                error("Can not create project for language $language")
-            }
+        logger.info("Start to create a new file for language \"${language.id}\"")
 
-            logger.info("Start to create new for language $language")
+        val file = AtomicReference<PsiFile>()
 
-            val file = AtomicReference<PsiFile>()
-            ApplicationManager.getApplication().invokeAndWait {
+        when (language.id) {
+            LanguageId.Python.name -> ApplicationManager.getApplication().invokeAndWait {
                 val psiFile = createPsiFile(project, "${language.id}_dummy", language, "")
                 file.set(psiFile)
             }
 
-            logger.info("New file for language has been successfully created")
-            SingleFileProject(language, project, file.get(), disposable)
+            LanguageId.kotlin.name -> ApplicationManager.getApplication().invokeAndWait {
+                val scope = GlobalSearchScope.allScope(project)
```

Comment:
I'd suggest moving the code for getting a virtual file to a separate file, and just calling the necessary function here. As we add more languages and more logic here, it will affect the size of this class. I mean to create e.g. a base interface and next two objects: for Python and Kotlin projects